src/Goteo/Entity/Invest/InvestOrigin.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Goteo\Entity\Invest;
+
+use Goteo\Model\Invest;
+
+class InvestOrigin
+{
+    private int $invest_id;
+    private string $source;
+    private string $detail;
+    private string $allocated;
```

Comment:
`$allocated` could be null, as the SQL doesn't state NOT NULL. So this property should be `?string`. Unless the SQL should be NOT NULL, in that case, the model would be already fine.